tools-v2/pkg/cli/command/curvebs/peer/utils/leader.go

```diff
@@ -0,0 +1,135 @@
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/proto/proto/cli2"
+	common "github.com/opencurve/curve/tools-v2/proto/proto/common"
+	"google.golang.org/grpc"
+)
+
+// LeaderRpc the rpc client for the rpc function GetLeader
+type LeaderRpc struct {
+	Info    *basecmd.Rpc
+	Request *cli2.GetLeaderRequest2
+	Cli     cli2.CliService2Client
+}
+
+var _ basecmd.RpcFunc = (*LeaderRpc)(nil) // check interface
+
+// NewRpcClient ...
+func (ufRp *LeaderRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	ufRp.Cli = cli2.NewCliService2Client(cc)
+}
+
+// Stub_Func ...
+func (ufRp *LeaderRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return ufRp.Cli.GetLeader(ctx, ufRp.Request)
+}
+
+// Configuration the configuration for peer.
+type Configuration struct {
+	Peers []*common.Peer
+}
+
+// Options rpc request options.
+type Options struct {
+	Timeout    time.Duration
+	RetryTimes int32
+}
+
+// ParsePeer parse the peer string
+func ParsePeer(peer string) (*common.Peer, error) {
+	cs := strings.Split(peer, ":")
+	if len(cs) != 3 {
+		return nil, fmt.Errorf("error format for the peer info %s", peer)
+	}
+	id, err := strconv.ParseUint(cs[2], 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error format for the peer id %s", cs[2])
+	}
+	cs = cs[:2]
+	address := strings.Join(cs, ":")
+	return &common.Peer{
+		Id:      &id,
+		Address: &address,
+	}, nil
+}
+
+// ParseConfiguration parse the conf string into Configuration
+func ParseConfiguration(conf string) (*Configuration, error) {
+	if len(conf) == 0 {
+		return nil, fmt.Errorf("empty group configuration")
+	}
+
+	confs := strings.Split(conf, ",")
+	configuration := &Configuration{}
+	for _, c := range confs {
+		peer, err := ParsePeer(c)
+		if err != nil {
+			return nil, err
+		}
+		configuration.Peers = append(configuration.Peers, peer)
+	}
+	return configuration, nil
+}
+
+// GetLeader get leader for the address.
+func GetLeader(logicalPoolID, copysetID uint32, conf Configuration, opts Options) (*common.Peer, error) {
+	if len(conf.Peers) == 0 {
+		return nil, errors.New("empty group configuration")
+	}
+	for _, peer := range conf.Peers {
+		rpcCli := &LeaderRpc{
+			Request: &cli2.GetLeaderRequest2{
+				LogicPoolId: &logicalPoolID,
+				CopysetId:   &copysetID,
+				Peer:        peer,
+			},
+			Info: basecmd.NewRpc([]string{peer.GetAddress()}, opts.Timeout, opts.RetryTimes, "GetLeader"),
+		}
+		response, errCmd := basecmd.GetRpcResponse(rpcCli.Info, rpcCli)
+		if errCmd.TypeCode() != cmderror.CODE_SUCCESS {
+			fmt.Printf("failed to acquire leader peer info %s error : %s", peer.GetAddress(), errCmd.Message)
+			continue
+		}
+		resp, ok := response.(*cli2.GetLeaderResponse2)
+		if !ok {
+			fmt.Printf("error type interface when acquire leader peer info %s", peer.GetAddress())
+			continue
+		}
+		if resp.Leader != nil {
+			return resp.Leader, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to acquire leader peer info")
+}
```

Comment:
Is this the command `curve bs query leader `?

Reply:
The old tool do not have this command,  and can get leader info by the command `curve_ops_tool check-copyset --logicalPoolId=1 --copysetId=21 --detail `

Reply:
> The old tool do not have this command, and can get leader info by the command `curve_ops_tool check-copyset --logicalPoolId=1 --copysetId=21 --detail `

In the new command, you can implement it according to the format of curve bs query copyset leader, and organize the files according to the structure of this command as above, but you can not add it as a subcommand of the corresponding command.